### PR TITLE
Fel 828

### PR DESCRIPTION
**Pull request by AlbyGiaco**

### CODE REVIEW

File: frontend/src/components/campaign/actions-section.tsx [hunk 1]
```diff
@@ -192,14 +198,27 @@ const ActionSection = ({ setTaskCompleted, actions, cid }: IProps) => {
...
+                )
+                  ? "opacity-50"
+                  : "hover:opacity-90"
               }`}
-          disabled={!isWhitelisted}
+          disabled={
+            !(isWhitelisted && actionsState.every((action: any) => action.done))
+          }
+          onClick={async () =>
+            setClaimlist(await addClaimlist(actionsState, cid))
```
Comment (Knos01) [inline frontend/src/components/campaign/actions-section.tsx:212]: Should the function be called "addToClaimlist"?

File: frontend/src/hooks/useTwitterAPI.ts [hunk 2]
```diff
@@ -20,6 +20,26 @@ export const useTwitterAPI = () => {
...
+      method: "post",
+      body: JSON.stringify({
+        actions,
+        account,
+        campaignId,
+        twitterHandle,
+        params: { next_token },
+      }),
+    };
+    const res = await fetch("/api/addClaimlist", opt);
```
Comment (Knos01) [inline frontend/src/hooks/useTwitterAPI.ts:40]: same as above

File: frontend/src/pages/api/addClaimlist.ts [hunk 3]
```diff
@@ -0,0 +1,192 @@
+import Airtable from "airtable";
```
Comment (Knos01) [inline frontend/src/pages/api/addClaimlist.ts:1]: same here for file name

File: frontend/src/hooks/useTwitterAPI.ts [hunk 4]
```diff
@@ -20,6 +20,26 @@ export const useTwitterAPI = () => {
     const res = await fetch("/api/queryTwitter", opt);
     return await res.json();
   };
+  const makeQueryClaimlist: any = async (
```
Comment (nicolas-racchi) [inline frontend/src/hooks/useTwitterAPI.ts:23]: this hook was intended for Twitter queries, can we place this function elsewhere to make things more clear? Maybe in hooks/useClaimCampaign where we currently only have the verifyAndClaim function?

Or even in a new hook file if you prefer!

File: frontend/src/pages/api/addClaimlist.ts [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+import axios from "axios";
+import { NextApiResponse } from "next";
+import { utils } from "ethers";
+const airtableKey = process.env.AIRTABLE_KEY;
+const twitterToken = process.env.TWITTER_TOKEN;
+
+export default async function handler(req: any, res: NextApiResponse) {
```
Comment (nicolas-racchi) [inline frontend/src/pages/api/addClaimlist.ts:8]: Can you add a short doc to the API README for this api call?

File: frontend/src/hooks/useTwitterAPI.ts [hunk 6]
```diff
@@ -20,6 +20,26 @@ export const useTwitterAPI = () => {
...
+    actions: any,
+    account: string,
+    campaignId: string,
+    twitterHandle: string,
+    next_token: string | null = null
+  ) => {
+    const opt: any = {
+      method: "post",
```
Comment (nicolas-racchi) [inline frontend/src/hooks/useTwitterAPI.ts:31]: Careful, I think POST needs to be capitalized, I'm not 100% sure but I think I had problems with this before

**Review by Knos01 — APPROVED** [review]

**Pull request merged**